tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/telemetryMetrics/util/TelemetryMetricID.java

```diff
@@ -30,59 +30,48 @@ public class TelemetryMetricID {
 
     public final String name;
     public final Attributes attributes;
+    public final TelemetryMetricDefinition.MetricType type;
 
-    public TelemetryMetricID(String classMethodName, Attributes attributes) {
+    public TelemetryMetricID(String classMethodName, TelemetryMetricDefinition.MetricType type, Attributes attributes) {
         this.name = classMethodName;
         this.attributes = attributes;
-
+        this.type = type;
     }
 
     @Override
     public String toString() {
         StringBuilder sb = new StringBuilder();
-        sb.append(name + " Attributes:");
-        for (AttributeKey<?> key : attributes.asMap().keySet()) {
-            sb.append("[" + key.toString() + "=" + attributes.asMap().get(key).toString() + "]");
+        sb.append(name);
+        sb.append(" Type: ").append(type);
+        sb.append(" Attributes: ");
+        for (Entry<AttributeKey<?>, Object> e : attributes.asMap().entrySet()) {
+            sb.append("[");
+            sb.append(e.getKey());
+            sb.append("=");
+            sb.append(e.getValue());
+            sb.append("]");
         }
         return sb.toString();
     }
 
     @Override
-    public boolean equals(Object o) {
-        TelemetryMetricID other = (TelemetryMetricID) o;
-
-        if (name != other.name) {
-            return false;
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
         }
-
-        if (attributes.size() != other.attributes.size()) {
+        if (obj == null) {
             return false;
         }
-
-        for (AttributeKey<?> key : attributes.asMap().keySet()) {
-            if (attributes.get(key) != other.attributes.get(key)) {
-                return false;
-            }
+        if (getClass() != obj.getClass()) {
+            return false;
         }
-
-        return true;
+        TelemetryMetricID other = (TelemetryMetricID) obj;
+        return Objects.equals(attributes, other.attributes) && Objects.equals(name, other.name) && type == other.type;
     }
 
     @Override
     public int hashCode() {
-        Map<?, ?> map = attributes.asMap();
-        Object[] names = map.entrySet().toArray();
-        Object[] value = map.keySet().toArray();
-
-        int namesHash = Arrays.deepHashCode(names);
-        int valuesHash = Arrays.deepHashCode(value);
-
-        int hash = 17;
-        hash = hash * 31 + namesHash;
-        hash = hash * 31 + valuesHash;
-        hash = hash * 31 + name.hashCode();
-
-        return hash;
+        return Objects.hash(attributes, name, type);
```

Comment:
I wasn't aware of this bit of syntactic sweetness. Thanks for making me aware of it :)

Reply:
I'll be honest, Eclipse's "Generate hashCode and equals" that made me aware of it.

Reply:
I should also note that this is only valid because implementations of `Attributes` are required to implement `equals` and `hashCode` correctly, otherwise the code that you had which was doing its own hashing of `attributes` would have been necessary.

Often interfaces assert this by redeclaring `equals` and `hashCode` (e.g. `java.util.Map` does this), but in the case of `Attributes`, it's only stated [in the class description](https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/1.39.0/io/opentelemetry/api/common/Attributes.html).